core/templates/pages/admin-page/features-tab/admin-features-tab.component.spec.ts

```diff
@@ -0,0 +1,372 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the feature tab in admin page.
+ */
+
+import { ComponentFixture, fakeAsync, async, TestBed, flushMicrotasks } from
+  '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { FormsModule } from '@angular/forms';
+
+import { AdminDataService } from '../services/admin-data.service';
+import { AdminFeaturesTabComponent } from './admin-features-tab.component';
+import { AdminPageData } from 'domain/admin/admin-backend-api.service';
+import { AdminTaskManagerService } from
+  '../services/admin-task-manager.service';
+import { PlatformFeatureAdminBackendApiService } from
+  'domain/platform_feature/platform-feature-admin-backend-api.service';
+import { PlatformParameterObjectFactory, FeatureStage, PlatformParameter } from
+  'domain/platform_feature/platform-parameter-object.factory';
+import { PlatformParameterFilterType, ServerMode } from
+  'domain/platform_feature/platform-parameter-filter-object.factory';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+
+describe('Admin page feature tab', function() {
```

Comment:
Suggest also making sure that all of the different structural changes happening in the prod code are being verified in tests below. We definitely don't want bugs in the admin panel leading to unexpected changes to feature flags that, in the worst case, actually pass validation and potentially cause immediate effects on users.

Reply:
Done, improved tests for them.